### PR TITLE
KUZ-344 [security] CRUD profiles

### PR DESCRIPTION
**Pull request by xbill82**
This covers the CRUDL methods for profiles. The main logic is situated in the SecurityController and the profileRepository.
Coming up... the update to the API documentation for [SecurityController](http://kuzzleio.github.io/kuzzle-api-documentation/#security-controller).

### CODE REVIEW

File: lib/api/controllers/securityController.js [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+   * @returns {Promise}
+   */
+  this.getRole = function (requestObject) {
+    kuzzle.pluginsManager.trigger('data:getRole', requestObject);
+
+    return kuzzle.repositories.role.loadRole(
+      kuzzle.repositories.role.getRoleFromRequestObject(requestObject)
+      )
+      .then(role => {
+        return Promise.resolve(new ResponseObject(requestObject, role || {}));
```
Comment (stafyniaksacha) [inline lib/api/controllers/securityController.js]: we have to use Q instead of Promise objects

Comment (AnthonySendra) [inline lib/api/controllers/securityController.js]: Promise works well for basic promise functionalities

**Comment by AnthonySendra** [comment]
Need to update in order to return `_source` attribute in case of a search.
You can do the same think as for Role

```
return kuzzle.repositories.role.searchRole(requestObject)
      .then((response) => {
        if (requestObject.data.body.hydrate && response.data.body && response.data.body.hits) {
          response.data.body.hits = response.data.body.hits.map(role => {
            var parsedRole = {};

            parsedRole._id = role._id;
            parsedRole._source = {indexes: role.indexes};

            return parsedRole;
          });
        }

        return Promise.resolve(response);
      });
```

When the user ask for hydrated object, we have to map the hits from ResponseObject for inject `_source` attribute.
This is not a perfect solution, but for Beta, it's could be enough, and next we can clean without impact on the API

File: lib/api/controllers/securityController.js (hunk 1 above)
Comment (xbill82) [inline lib/api/controllers/securityController.js]: Ok so, what do we stick to? I've seen some comments on #123 telling that promised returns should be harmonized on `q(response)`.

**Comment by xbill82** [comment]
@AnthonySendra the response I get for a `profiles/_search` is

```
"result": {
    "_shards": {
      "failed": 0,
      "successful": 5,
      "total": 5
    },
    "hits": [
      {
        "_id": "my-profile-1",
        "_index": "%kuzzle",
        "_score": 1,
        "_source": {
          "_id": "my-profile-1",
          "roles": [
            "role1"
          ]
        },
        "_type": "profiles"
      },
      {
        "_id": "my-profile-2",
        "_index": "%kuzzle",
        "_score": 1,
        "_source": {
          "_id": "my-profile-2",
          "roles": [
            "role1"
          ]
        },
        "_type": "profiles"
      }
    ],
    "max_score": 1,
    "timed_out": false,
    "took": 91,
    "total": 2
  },
```

It seems to already work for me, no?

**Comment by AnthonySendra** [comment]
@xbill82 Did you try to call the same url but with `hydrate` to `true` ?

**Comment by xbill82** [comment]
@AnthonySendra whooops...

**Comment by xbill82** [comment]
@AnthonySendra ok, does this look better?

```
"result": {
    "hits": [
      {
        "_id": "my-profile-1",
        "_source": {
          "roles": [
            {
              "_id": "role1",
              "closures": {},
              "indexes": {
                "fakeIndex1": {
                  "collections": {
                    "fakeCollection1": {
                      "controllers": {
                        "fakeController1": {
                          "actions": {
                            "fakeAction1": true
                          }
                        }
                      }
                    }
                  }
                }
              }
            }
          ]
        }
      },
      {
        "_id": "my-profile-2",
        "_source": {
          "roles": [
            {
              "_id": "role1",
              "closures": {},
              "indexes": {
                "fakeIndex1": {
                  "collections": {
                    "fakeCollection1": {
                      "controllers": {
                        "fakeController1": {
                          "actions": {
                            "fakeAction1": true
                          }
                        }
                      }
                    }
                  }
                }
              }
            }
          ]
        }
      }
    ],
    "total": 2
  },
```

(looks pretty ugly, btw)

**Comment by AnthonySendra** [comment]
@xbill82 the role inside `_source` is not well formated, but try to merge my branch KUZ-342-crudl-role, I have made some update that will propably fix the problem

File: features/support/apiREST.js [hunk 2]
```diff
@@ -1,6 +1,7 @@
 var
   config = require('./config')(),
   rp = require('request-promise'),
+  apiVersion = require('../../package.json').apiVersion;
```
Comment (scottinet) [inline features/support/apiREST.js]: overwritten by the next line

Comment (xbill82) [inline features/support/apiREST.js]: :+1: must be a bad merge, thx!

**Comment by xbill82** [comment]
@AnthonySendra I've reformatted the hydrated results. Can you validate please?

```
"result": {
    "hits": [
      {
        "_id": "my-profile-2",
        "_source": {
          "roles": [
            {
              "_id": "role1",
              "_source": {
                "indexes": {
                  "fakeIndex1": {
                    "collections": {
                      "fakeCollection1": {
                        "controllers": {
                          "fakeController1": {
                            "actions": {
                              "fakeAction1": true
                            }
                          }
                        }
                      }
                    }
                  }
                }
              }
            }
          ]
        }
      }
    ],
    "total": 1
  }
```

**Pull request merged**